src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5567,6 +5564,25 @@ private SqlFedAuthToken getFedAuthToken(SqlFedAuthInfo fedAuthInfo) throws SQLSe
 
                 // Break out of the retry loop in successful case.
                 break;
+            } else if (authenticationString.equalsIgnoreCase(SqlAuthentication.DefaultAzureCredential.toString())) {
+                if (!msalContextExists()) {
```

Comment:
I know this was there before, but can we just refactor this so we don't have to dup this now 5 times